tool_sideview/sideview_view.py

```diff
@@ -1621,21 +1618,22 @@ def get_layer(spatialite_path, table_name, geom_column=""):
             # geom = line.get('geom', QgsGeometry.fromPolyline([p1, p2]))
 
             feat.setGeometry(geom)
-
+            channel_id = line.get("channel_id", None),
+            # Casting ids to strings is needed due to issue with casting values in memory layers in QGIS < 3.16.6
             feat.setAttributes(
                 [
                     i,
-                    line["id"],
+                    str(line["id"]),
                     line["type"],
-                    line["start_node"],
-                    line["end_node"],
+                    str(line["start_node"]),
+                    str(line["end_node"]),
                     line.get("start_node_idx", None),
                     line.get("end_node_idx", None),
                     line.get("start_level", None),
                     line.get("end_level", None),
                     line.get("start_height", None),
                     line.get("end_height", None),
-                    line.get("channel_id", None),
+                    str(channel_id) if channel_id is not None else channel_id,
```

Comment:
The effect is the same, but I think it is a bit more explicit to use None instead of channel_id:

```suggestion
                    str(channel_id) if channel_id is not None else None,
```